src/main/kotlin/org/rust/ide/refactoring/introduceVariable/impl.kt

```diff
@@ -70,9 +71,12 @@ private class ExpressionReplacer(
                 binding
             } else {
                 val parentLambda = chosenExpr.ancestorStrict<RsLambdaExpr>()
+                val lambdaMayBeAnchor = parentLambda != null && sortedExprs.all { parentLambda.isAncestorOf(it) }
                 val binding = introduceLet(anchor, let)
                 sortedExprs.forEach { it.replace(name) }
-                binding?.moveIntoLambdaBlockIfNeeded(parentLambda)
+                if (lambdaMayBeAnchor) {
+                    binding.moveIntoLambdaBlockIfNeeded(parentLambda)
+                } else binding
```

Comment:
[minor] we use code style where `if/else` braces should presence/absence both simultenuously

```suggestion
                } else {
                    binding
                }
```